nucliadb/nucliadb/common/datamanagers/labels.py

```diff
@@ -68,13 +141,30 @@ async def set_labels(self, kbid: str, labels: kb_pb2.Labels) -> None:
         Set all labels for a knowledge box (may include multiple labelsets)
         """
         async with self.driver.transaction() as txn:
+            labelset_ids = list(labels.labelset.keys())
+            await LabelsDataManager._add_to_labelset_ids(kbid, txn, labelset_ids)
```

Comment:
What about removal? This is an overwrite right? Correct me if I'm wrong but only adding here might mean we aren't removing things?

Reply:
good catch!